app/models/concerns/foreman_bootdisk/orchestration/compute.rb

```diff
@@ -0,0 +1,82 @@
+require 'tmpdir'
+
+module ForemanBootdisk
+  module Orchestration
+    module Compute
+      extend ActiveSupport::Concern
+
+      included do
+        after_validation :queue_bootdisk_compute
+        after_build :rebuild_with_bootdisk
+      end
+
+      def bootdisk_isofile
+        File.join(Dir.tmpdir, "#{name}.iso")
+      end
+
+      def queue_bootdisk_compute
+        return unless compute? && errors.empty? && new_record?
+        return unless provision_method == 'bootdisk'
+        queue.create(name: _('Generating iso image for %s') % self, priority: 5,
+                     action: [self, :setGenerateIsoImage])
+        queue.create(name: _('Upload iso image to datastore for %s') % self, priority: 6,
+                     action: [self, :setIsoImage])
+        queue.create(name: _('Attach iso image to cdrom drive for %s') % self, priority: 7,
+                     action: [self, :setAttachIsoImage])
+      end
+
+      def bootdisk_generate_iso_image
+        ForemanBootdisk::ISOGenerator.generate(ipxe: bootdisk_template_render, dir: Dir.tmpdir) do |image|
+          FileUtils.mv image, bootdisk_isofile
+        end
+      end
+
+      def bootdisk_upload_iso
+        compute_resource.iso_upload(bootdisk_isofile, uuid)
+      end
+
+      def bootdisk_attach_iso
+        compute_resource.iso_attach(File.basename(bootdisk_isofile), uuid)
+      end
+
+      def setGenerateIsoImage
+        logger.info 'Generating iso image for %s' % name
+        bootdisk_generate_iso_image
+      rescue => e
+        failure _("Failed to generate iso image for instance %{name}: %{message}\n ") % { name: name, message: e.message }, e
```

Comment:
These failure strings (here and below) probably shouldn't have trailing whitespace as it causes translators problems, remove the `\n `.